File: app/src/main/java/com/omar/retromp3recorder/app/usecase/ChangeBitrateUC.java
package com.omar.retromp3recorder.app.usecase;

import com.omar.retromp3recorder.app.recorder.VoiceRecorder;
import com.omar.retromp3recorder.app.repo.BitRateRepo;

import javax.inject.Inject;

import io.reactivex.Completable;

public class ChangeBitrateUC {
    private final BitRateRepo repo;

    @Inject
    public ChangeBitrateUC(BitRateRepo repo) {
        this.repo = repo;
    }

    public Completable execute(VoiceRecorder.BitRate bitRate) {
        return Completable.fromAction(() -> repo.newValue(bitRate));
    }
}

File: app/src/main/java/com/omar/retromp3recorder/app/usecases/ChangeBitrateUC.kt
package com.omar.retromp3recorder.app.usecases

import com.omar.retromp3recorder.recorder.Mp3VoiceRecorder
import io.reactivex.Completable
import javax.inject.Inject

class ChangeBitrateUC @Inject constructor(private val repo: com.omar.retromp3recorder.state.BitRateRepo) {
    fun execute(bitRate: Mp3VoiceRecorder.BitRate): Completable {
        return Completable.fromAction { repo.newValue(bitRate) }
    }
}
